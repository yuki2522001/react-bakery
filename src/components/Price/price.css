.price {
  color: var(--text-primary);
  font-weight: bold;
  margin: 0;
}