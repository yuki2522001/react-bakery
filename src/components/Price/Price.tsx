import React, { memo } from "react";
import "./price.css";

interface PriceProps {
  value?: number;
  className?: string;
}

const Price: React.FC<PriceProps> = memo(({ value="", className }) => {
  return (
    <p data-testid="price" className={className}>
      ${value}
    </p>
  );
});

export default Price;
