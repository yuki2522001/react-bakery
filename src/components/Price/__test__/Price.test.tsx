import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Price from "../Price";

describe("Price component", () => {
  test("should render price component", () => {
    const { getByTestId } = render(<Price value={0} />);
    expect(getByTestId("price")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Price value={0} />);
    expect(asFragment()).toMatchSnapshot();
  });
})