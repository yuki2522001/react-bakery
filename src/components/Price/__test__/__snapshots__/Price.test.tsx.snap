// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component matches snapshot 1`] = `
<DocumentFragment>
  <p
    data-testid="price"
  >
    $0
  </p>
</DocumentFragment>
`;
